assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeed.js

```diff
@@ -165,6 +166,9 @@ export default function DashboardPageSpeed() {
 				{ dataSrc === DATA_SRC_LAB && <LabReportMetrics data={ reportData } error={ reportError } /> }
 				{ dataSrc === DATA_SRC_FIELD && <FieldReportMetrics data={ reportData } error={ reportError } /> }
 			</section>
+			{ ! reportError && (
+				<Recommendations referenceURL={ referenceURL } strategy={ strategy } />
+			) }
```

Comment:
The source link (`ReportDetailsLink`) and the "Run test again" link (for Lab Data only) should for now remain at the bottom of the overall widget, which currently is not the case, since the new `Recommendations` component was added to the very bottom. So we need to move this section out of the `LabReportMetrics` / `FieldReportMetrics` components and display it once at the bottom of the overall widget (`DashboardPageSpeed`). We can then only display the "Run test again" link if `dataSrc === DATA_SRC_LAB`.

Reply:
Updated.